### ISSUE TITLE
guideline for barcode selection

### ISSUE THREAD

**Issue by hjack123** [r3fang/SnapATAC#93]
Hi,

Wondering if there is a guideline setting the UMI and promoter ration thresholds for barcodes selection? Thanks!!

**Comment by r3fang** [comment]
Hello. 

The selection of cutoff is quite subjective and maybe sample specific. Usually the high-quality population will be located at the upper right corner of this plot, separating from the low quality barcodes. you might have to choose the cutoff by eyes. The cutoff I find is quite useful is 3.5-5 for logUMI and 0.15 to 0.8 for promoter ratio.

**Comment by hjack123** [comment]
Thank you very much for your reply!!

**Comment by hjack123** [comment]
tried googling but couldnt find Scrumblet, would you mind sharing the link? thanks

**Comment by r3fang** [comment]
Here you go

https://www.cell.com/cell-systems/pdfExtended/S2405-4712(18)30474-5 <https://www.cell.com/cell-systems/pdfExtended/S2405-4712(18)30474-5>

--
Rongxin Fang, Ren Lab
Ludwig Cancer Research
Bioinformatics Ph.D. Student
University of California, San Diego

**Comment by hjack123** [comment]
Thank u so much, out of curiosity, would i be able to do the filtering barcode with the QC described here (https://satijalab.org/signac/articles/pbmc_vignette.html) and continue ur pipeline?

**Comment by r3fang** [comment]
Yes, if you have qc file from 10X cell ranger, you can filter cells based on their QC report

**Comment by hjack123** [comment]
Thank you 
 I have been stuck here for a while, trying to combine 4 snap files
>x.sp = snapRbind(x.landmark.sp, x.query.sp);

is there a way to check? buy looking at the context of x.landmark.sp and x.query.sp, they look the same to me.
something related to the note
Note: To merge snap objects, all the matrix (bmat, gmat, pmat) and metaData must be of the same number of columns between snap objects.

**Comment by r3fang** [comment]
Can you let me know the error message?

**Comment by hjack123** [comment]
> x.sp = snapRbind(x.landmark.sp, x.query.sp);
Error in rbind(deparse.level, ...) : 
  numbers of columns of arguments do not match

**Comment by hjack123** [comment]
The problem is solved.
btw, there might be a typo here
x.after.sp = runHarmony(
    obj=x.sp, 
    eigs.dims=1:22, 
    meta_data=x.sp@sample # sample index
  );

eigs.dim

**Comment by hjack123** [comment]
if using umap creates an outlier which shows on the plot. is there a way to change xlim and ylim so the plot don't include the outlier?

**Comment by JBreunig** [comment]
> Thank you
> I have been stuck here for a while, trying to combine 4 snap files
> 
> > x.sp = snapRbind(x.landmark.sp, x.query.sp);
> 
> is there a way to check? buy looking at the context of x.landmark.sp and x.query.sp, they look the same to me.
> something related to the note
> Note: To merge snap objects, all the matrix (bmat, gmat, pmat) and metaData must be of the same number of columns between snap objects.

hjack123...does this Note refer to your solution to the problem?  I'm having the same issue at this step. I can merge 2 samples from a new dataset but adding a 3rd from and older dataset causes the error you saw.

**Comment by JBreunig** [comment]
> > Thank you
> > I have been stuck here for a while, trying to combine 4 snap files
> > > x.sp = snapRbind(x.landmark.sp, x.query.sp);
> > 
> > 
> > is there a way to check? buy looking at the context of x.landmark.sp and x.query.sp, they look the same to me.
> > something related to the note
> > Note: To merge snap objects, all the matrix (bmat, gmat, pmat) and metaData must be of the same number of columns between snap objects.
> 
> hjack123...does this Note refer to your solution to the problem? I'm having the same issue at this step. I can merge 2 samples from a new dataset but adding a 3rd from and older dataset causes the error you saw.

Solved...just for future users my issue was caused by there being an extra metaData column in the newer CellRanger outputs.  I just made the new column in metaData NULL to match the old metadata type number (19).

> # remove NO BAROCDE line
>     barcodes = barcodes[2:nrow(barcodes),];
>     **barcodes$peak_region_cutsites = NULL  ###key to old versus new samples in cell ranger**
>     barcodes$logUMI = log10(barcodes$passed_filters + 1);

**Comment by TheSallyGardens** [comment]
> if using umap creates an outlier which shows on the plot. is there a way to change xlim and ylim so the plot don't include the outlier?
Hi！hjack123
Did you solve the problem？

**Comment by hjack123** [comment]
I ended up removing those cells (by filtering on umap locations) and re-plotting. same visualization purpose as changing xylim. hope it helps
